apps/condo/domains/miniapp/schema/B2BAppAccessRight.test.js

```diff
@@ -186,5 +203,327 @@ describe('B2BAppAccessRight', () => {
+        expect(propertyByUser.id).toBe(property.id)
+
+        // add permissions for B2BApp
+        await updateTestB2BAppAccessRight(admin, right.id, {
+            canReadProperty: true,
+            canManageProperty: true,
+            canReadOrganization: true, // required permission for manage property
+        })
+
+        // B2BApp with permissions
+        await createTestProperty(serviceUserClient, organization)
+
+        const countByUser2 = await Property.count(user, { organization: { id: organization.id } })
+        expect(countByUser2).toBe(2)
+
+        const countWithPermissions = await Property.count(serviceUserClient, {})
+        expect(countWithPermissions).toBe(2)
+        const propertyWithPermissions = await Property.getOne(serviceUserClient, { id: property.id })
+        expect(propertyWithPermissions.id).toBe(property.id)
+
+        await updateTestProperty(serviceUserClient, property.id, {})
+        await Property.softDelete(serviceUserClient, property.id)
+
+        const countByUser3 = await Property.count(user, { organization: { id: organization.id } })
+        expect(countByUser3).toBe(1)
+    })
+
+    test('Contact', async () => {
+        const [organization] = await registerNewOrganization(user)
+
+        const [app] = await createTestB2BApp(admin)
+        await createTestB2BAppContext(admin, app, organization, { status: 'Finished' })
+        const [right] = await createTestB2BAppAccessRight(admin, serviceUser, app)
+
+        const [property] = await createTestProperty(user, organization)
+        const [contact] = await createTestContact(user, organization, property)
+
+        // B2BApp without permissions
+        const countWithoutPermissions = await Contact.count(serviceUserClient, {})
+        expect(countWithoutPermissions).toBe(0)
+        const contactWithoutPermissions = await Contact.getOne(serviceUserClient, { id: contact.id })
+        expect(contactWithoutPermissions).toBeUndefined()
+
+        await expectToThrowAccessDeniedErrorToObj(async () => {
+            await createTestContact(serviceUserClient, organization, property)
+        })
+        await expectToThrowAccessDeniedErrorToObj(async () => {
+            await updateTestContact(serviceUserClient, contact.id, {})
+        })
+        await expectToThrowAccessDeniedErrorToObj(async () => {
+            await Contact.softDelete(serviceUserClient, contact.id)
+        })
+
+        const countByUser = await Contact.count(user, {})
+        expect(countByUser).toBe(1)
+        const contactByUser = await Contact.getOne(user, { id: contact.id })
+        expect(contactByUser.id).toBe(contact.id)
+
+        // add permissions for B2BApp
+        await updateTestB2BAppAccessRight(admin, right.id, {
+            canReadContact: true,
+            canManageContact: true,
+            canReadOrganization: true, // required permission for manage contact
+            canReadProperty: true, // required permission for manage contact
```

Comment:
Did I guess it right, that each miniapp should resolve it manually?

Reply:
Do we want to build a relation-deps graph or we'll resolve it in UI-state?

Reply:
Only the admin and the support can change permissions. A miniapp itself cannot change permissions.

There are no snippets yet. But it would be convenient